include/sempr/entity/spatial/Geometry.hpp

```diff
@@ -40,15 +40,7 @@ class Geometry : public Entity
     // unable to override if method is const.
     // its not allow to use this geometry in another geometry (e.g. collection). Otherwise you will get multiple ownerships to this object!
     // To use a collection build a clone of this geometry or construct a new one.
-    virtual const geom::Geometry* geometry();
-
-    /**
-     * @brief Set the GEOS GEOM geometry object.
-     * The new object have to be created by the GeometryFactory!
-     * 
-     * @param geometry The new geometry object to set.
-     */
-    void setGeometry(geom::Geometry* geometry);
+    virtual const geom::Geometry* getGeometry();
```

Comment:
1. Why do you need `getGeometry()` _and_ `geometry()`?
2. The comment `// unable to override if method is const.` -- is that leftover from my code? 'Cause I tend to disagree. :wink:

Reply:
There are two Get methods because of this:
 - The user shall only get a const point to avoid ownership issues by the use of geos
 - Otherwise the Geometry base class need a non-const pointer of the geometry to apply filter and since the pointer is hold in the child classes the base class can only get access via a virtual method.

Reply:
I think you switched things up a bit: Now the non-const method returns a pointer to const, and the const method returns a pointer to non-const! :sweat_smile:

Reply:
It was a work around to fix the strange best match behavior of the compiler.
In dcadddeb596ea4d3c0ae96020c2bbeefad3de695 its corrected by separate names for the getter.